packages/financial-templates-lib/src/logger/DiscordTransport.ts

```diff
@@ -10,6 +10,11 @@ export class DiscordTransport extends Transport {
   private readonly defaultWebHookUrl: string | string[];
   private readonly escalationPathWebhookUrls: { [key: string]: string | string[] };
   private readonly postOnNonEscalationPaths: boolean;
+
+  private logQueue: any = [];
```

Comment:
It looks like the types for this queue are pretty well-defined. Might make sense to just make it explicit.

```suggestion
  interface Body {
    username: string;
    avatar_url: string;
    embeds: { title: string; description: string; color: number }[];
  }
  interface QueueElement {
    body: Body;
    webHook: string;
  };

  private logQueue: QueueElement[] = [];
```

Reply:
I had not done this as the shift operator does not play well with this:
<img width="584" alt="image" src="https://user-images.githubusercontent.com/12886084/159867142-0dde7f00-1598-4f2b-88d2-30546d2e912b.png">

Reply:
I'll try find a workaround.

Reply:
I had to add a ts-ignore for this. could not find another way.

Reply:
then I had to add `@typescript-eslint/ban-ts-comment` to get this through the linter :(

Reply:
As mentioned IRL, I think you can use the non-null assertion operator, `!` since you can't _prove_ to the compiler that it's non-null.

```ts
({ webHook, body } = this.logQueue.shift()!);
```